analyses/snv-callers/util/wrangle_functions.R

```diff
@@ -0,0 +1,260 @@
+################################################################################
+
+calculate_vaf <- function(maf_df) {
+  # Creates these new variables from a MAF formatted data.frame provided: VAF,
+  # mutation_id, base_change, change.
+  #
+  # Args:
+  #   maf_df: a maf formatted data.frame
+  #
+  # Returns:
+  #   a data.frame with all the original information in the `@data` part of the
+  #   maf object but with these new variables: VAF, mutation_id, base_change,
+  #   change, coding.
+
+  # Extract the data part of the maf object, put it through a dplyr pipe.
+  maf_df %>%
+    dplyr::mutate(
+      # Calculate the variant allele frequency
+      vaf = as.numeric(t_alt_count) / (as.numeric(t_ref_count) +
+        as.numeric(t_alt_count)),
+      # Create a base_change variable
+      base_change = paste0(Reference_Allele, ">", Allele),
+
+      # Create a numeric portion of the PolyPhen score
+      PolyPhen_numeric = as.numeric(stringr::word(PolyPhen, 2, sep = "\\(|\\)")),
+
+      # Create a categorical portion of the PolyPhen score
+      PolyPhen_category = stringr::word(PolyPhen, 1, sep = "\\("),
+
+      Variant_Classification = as.factor(Variant_Classification)
+    ) %>%
+    dplyr::mutate(
+      # From the base_change variable, summarize insertions, deletions, and
+      # changes that are more than one base into their own groups.
+      change = dplyr::case_when(
+        grepl("^-", base_change) ~ "ins",
+        grepl("-$", base_change) ~ "del",
+        nchar(base_change) > 3 ~ "long_change",
+        TRUE ~ base_change
+      )
+    ) %>%
+    dplyr::mutate(
+      # Create the mutation id based on the change variable as well as the
+      # gene symbol, start position, and sample ID.
+      mutation_id = paste0(
+        Hugo_Symbol, "_",
+        change, "_",
+        Start_Position, "_",
+        Tumor_Sample_Barcode
+      ),
+    )
+}
+
+maf_to_granges <- function(maf_df) {
+  # Turn MAF data.frame into a GRanges object. All of the original data.frame will
+  # be stored in the `mcols` slot of the GRanges object.
+  #
+  # Args:
+  #   maf_df: A MAF formatted data.frame with `Chromosome`, `Start_Position`,
+  #           `End_Position`, and `Strand`.
+  #
+  # Create the GRanges object
+  GenomicRanges::GRanges(
+    seqnames = maf_df$Chromosome,
+    ranges = IRanges::IRanges(
+      start = maf_df$Start_Position,
+      end = maf_df$End_Position
+    ),
+    strand = maf_df$Strand,
+    mcols = maf_df
+  )
+}
+
+wxs_bed_filter <- function(maf_df, wxs_bed = NULL, bp_window = 0) {
+  # Given a MAF formatted data.frame and a BED regions data.frame; filter out
+  # any variants of the MAF df that are not within the BED regions.
+  #
+  # Args:
+  #   maf_df: maf data that has been turned into a data.frame. Can be a maf object
+  #           that is subsetted using `@data`.
+  #   wxs_bed: a data.frame that has the windows to be used for TMB calculation.
```

Comment:
I don't have the exact BED regions files yet, but the plan is to have a script that sets them up and now I'm changing this argument so that it is a file.path to that BED file.